Add documentation for Spark and Flink new features (#679)
lightbend/cloudflow: docs/shared-content-source/docs/modules/develop/pages/use-spark-streamlets.adoc

* The name used in the blueprint for the specific instance of this streamlet being run.
* A function `checkpointDir` which returns a directory on a persistent storage where checkpoints can be safely kept, making them available across restarts.

== Controlling Streaming Query execution

> Spark Structured streaming provides mechanism for controlling timing of streaming data processing in the form of

Reviewer: ```suggestion
Spark Structured streaming provides a mechanism for controlling timing of streaming data processing in the form of
```

> Spark Structured streaming provides mechanism for controlling timing of streaming data processing in the form of

Reviewer 2: controlling "the" timing

> Spark Structured streaming provides a mechanism for controlling timing of streaming data processing in the form of

Reviewer: ```suggestion
Spark Structured streaming provides a mechanism for controlling the timing of streaming data processing in the form of
```

https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html#triggers[triggers].

> The trigger settings of a streaming query defines whether the query is going to be executed as micro-batch query with a fixed

Reviewer: ```suggestion
The trigger settings of a streaming query define whether the query is going to be executed as micro-batch query with a fixed
```

> The trigger settings of a streaming query defines whether the query is going to be executed as micro-batch query with a fixed

Reviewer 2: "define" instead of "defines" because it refers to settings. as "a" micro-batch query ...

> batch interval or as a continuous processing query. Different kinds of triggers that are supported by Spark are https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html#triggers[here]

Reviewer 2: [here].

> batch interval or as a continuous processing query. Different kinds of triggers that are supported by Spark are https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html#triggers[here]

Reviewer: ```suggestion
batch interval or as a continuous processing query. Different kinds of triggers that are supported by Spark are https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html#triggers[here].
```

> 'writeStream' method in stream logic allows to specify trigger used for Streamlet execution.

Reviewer 2: The 'writeStream' method in stream logic allows you to specify (not sure whether it should be a or the?) trigger ....

> 'writeStream' method in stream logic allows to specify trigger used for Streamlet execution.

Reviewer: ```suggestion
The `writeStream` method in stream logic allows to specify trigger used for Streamlet execution.
```
